TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.DeclarativesContext.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections.Generic;
+
+namespace TypeCobol.Analysis.Cfg
+{
+    public partial class ControlFlowGraphBuilder<D>
+    {
+        /// <summary>
+        /// Declarative context class.
+        /// </summary>
+        internal class DeclarativesContext
+        {
+            /// <summary>
+            /// Current Block before all declaratives sections and paragraphs.
+            /// </summary>
+            internal BasicBlockForNode CurrentBlock;
+
+            /// <summary>
+            /// All sections inside this Declaratives.
+            /// </summary>
+            internal LinkedList<CfgSectionSymbol> Sections;
+
+            /// <summary>
+            /// The related CFG builder
+            /// </summary>
+            internal ControlFlowGraphBuilder<D> Builder;
+
+            /// <summary>
+            /// Constructor
+            /// </summary>
+            /// <param name="currentProgramCfgBuilder">The related CFG Builder</param>
+            internal DeclarativesContext(ControlFlowGraphBuilder<D> currentProgramCfgBuilder)
+            {
+                Sections = new LinkedList<CfgSectionSymbol>();
+                Builder = currentProgramCfgBuilder;
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="currentBlock"></param>
+            internal void Start(BasicBlockForNode currentBlock)
+            {
+                CurrentBlock = currentBlock;
+            }
+            internal void AddSection(CfgSectionSymbol section)
+            {
+                Sections.AddLast(section);
+            }
+
+            /// <summary>
+            /// End the declaratives sections.
+            /// Each Section becomes a branch from the current block.
+            /// </summary>
+            internal void End(BasicBlockForNode nextBlock)
+            {
+                System.Diagnostics.Debug.Assert(Builder != null);
+                System.Diagnostics.Debug.Assert(CurrentBlock != null);
+                System.Diagnostics.Debug.Assert(nextBlock != null);
+
+                //First Current block is linked to the next block.
+                int nbIndex = Builder.Cfg.SuccessorEdges.Count;
+                Builder.Cfg.SuccessorEdges.Add(nextBlock);
+                CurrentBlock.SuccessorEdges.Add(nbIndex);
+
+                //For each section, link the current block to the first block of the section.
+                bool bFirstsection = true;
+                foreach (var section in Sections)
```

Comment:
Iterate over `Sections.Skip(1)` instead.

Reply:
Happy for _LinkQers_

Reply:
Yes I like Linq 😄 , but is is also a matter of readability. It took me a bit of time to understand we  are iterating over all sections but first and using only first sentence of each.